src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/IDependencyModelEqualityComparer.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    internal sealed class IDependencyModelEqualityComparer : IEqualityComparer<IDependencyModel>
+    {
+        public static IDependencyModelEqualityComparer Instance { get; } = new IDependencyModelEqualityComparer();
+
+        public bool Equals(IDependencyModel x, IDependencyModel y)
+        {
+            Requires.NotNull(x, nameof(x));
```

Comment:
Aren't Equals methods supposed to deal with nulls gracefully?

Reply:
True. It'll never happen in current usages, but I've updated it all the same.